### PR TITLE
feat: Unknown lookup status of a path in certificate yields AgentError

### PR DESCRIPTION
**Pull request by mraszyk**
# Description

Absence proofs have been recently enabled on the IC (see [MR](https://github.com/dfinity/ic/commit/2ac0fe91cd737977bf039c55502a31b363884c69)) and thus the IC is going to produce certificate whose lookup returns `Absent` for paths whose values are missing in the state tree. This PR reflects this change in the agent.

### CODE REVIEW

File: ic-agent/src/agent/response_authentication.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ pub(crate) fn lookup_request_status<Storage: AsRef<[u8]>>(
     ];
     match certificate.tree.lookup_path(&path_status) {
         LookupResult::Absent => Ok(RequestStatusResponse::Unknown),
-        LookupResult::Unknown => Ok(RequestStatusResponse::Unknown),
+        LookupResult::Unknown => Err(LookupPathUnknown(path_status.to_vec())),
```
Comment (ericswanson-dfinity) [inline ic-agent/src/agent/response_authentication.rs:70]: To see if I understand:
- Before https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/13886, IC would return LookupResult::Unknown
- After, IC will return LookupResult::Absent

Note that Agent::wait() will exit immediately if lookup_request_status returns Err(LookupPathUnknown).  Does this mean we need to wait to publish an agent version with this change until the IC is running a replica that incorporates MR13886?

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]

File: ic-agent/src/agent/response_authentication.rs (hunk 1 above)
Comment (mraszyk) [reply ic-agent/src/agent/response_authentication.rs:70]: Indeed, the above two bullet points are correct.

> we need to wait to publish an agent version with this change until the IC is running a replica that incorporates MR13886

yes, indeed

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]
Looks good.  Let's hold off on merging this until the relevant replica update has been executed.  Otherwise, we wouldn't be able to release agent-rs in the meantime with other changes.

**Comment by mraszyk** [comment]
The replica [release](https://dashboard.internetcomputer.org/release/3bcccef07408921fe849c92dd2437adc157ef9c3) containing absence proofs has been rolled out on all subnets. So we can proceed with this MR.

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**